shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java

```diff
@@ -216,6 +221,104 @@ private BindingTableRule createBindingTableRule(final String bindingTableGroup)
         return result;
     }
     
+    private void checkSameBindingTables(final Collection<String> bindingTableGroups, final Map<String, BindingTableRule> bindingTableRules) {
+        for (String each : bindingTableGroups) {
+            List<String> bindingTableList = Splitter.on(",").trimResults().splitToList(each.toLowerCase());
+            TableRule sampleTableRule = null;
+            for (String bindingTable : bindingTableList) {
+                TableRule tableRule = bindingTableRules.get(bindingTable).getTableRules().get(bindingTable);
+                if (null == sampleTableRule) {
+                    sampleTableRule = tableRule;
+                } else {
+                    checkSameActualDatasourceNamesAndActualTableIndex(sampleTableRule, tableRule, each);
+                }
+            }
+        }
+    }
+    
+    private void checkSameActualDatasourceNamesAndActualTableIndex(final TableRule sampleTableRule, final TableRule tableRule, final String bindingTableGroup) {
+        if (!CollectionUtils.isEqualCollection(sampleTableRule.getActualDatasourceNames(), tableRule.getActualDatasourceNames())) {
```

Comment:
@cheese8 Can we remove this util class? Replace it with `sampleTableRule.getActualDatasourceNames().equals( tableRule.getActualDatasourceNames()))`

Reply:
OK